version: '3.8'

services:
  db:
    container_name: "pyos-mysql"
    image: mysql:8.0
    command: --default-authentication-plugin=mysql_native_password
    restart: always
    ports:
      - "3306:3306"
    environment:
      MYSQL_ROOT_PASSWORD: minios
      MYSQL_DATABASE: minios
      MYSQL_USER: minios
      MYSQL_PASSWORD: minios
    volumes:
      - ./mysql:/docker-entrypoint-initdb.d
    networks:
      - minios_network

  django:
    container_name: "pyos-django"
    restart: on-failure
    build: ./
    command: bash -c "python manage.py makemigrations && python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    volumes:
      - ./:/app
      - static_volume:/app/static
    ports:
      - "8000:8000"
    env_file:
      - ./.env
    depends_on:
      - db
    networks:
      - minios_network

  vite:
    container_name: "pyos-vue"
    build:
      context: ./vue/mini_os
      dockerfile: ./Dockerfile
    volumes:
      - ./nginx/default.conf:/etc/nginx/conf.d/default.conf
    ports:
      - "3000:80"
    depends_on:
      - django
    networks:
      - minios_network


volumes:
  db_data:
  static_volume:

networks:
  minios_network: